test/vpc_component_test.go

```diff
@@ -0,0 +1,147 @@
+	assert.NoError(s.T(), err)
+	assert.Equal(s.T(), 0, len(nats))
+}
+
+func (s *ComponentSuite) TestPublicVPC() {
+	const component = "vpc/public"
+	const stack = "default-test"
+	const awsRegion = "us-east-2"
+
+	defer s.DestroyAtmosComponent(s.T(), component, stack, nil)
+	options, _ := s.DeployAtmosComponent(s.T(), component, stack, nil)
+	cidrBlock := atmos.Output(s.T(), options, "vpc_cidr")
+	assert.Equal(s.T(), "172.16.0.0/16", cidrBlock)
+
+	vpcId := atmos.Output(s.T(), options, "vpc_id")
+	require.True(s.T(), strings.HasPrefix(vpcId, "vpc-"))
+
+	vpc := aws.GetVpcById(s.T(), vpcId, awsRegion)
+
+	assert.Equal(s.T(), vpc.Name, fmt.Sprintf("eg-default-ue2-test-vpc-terraform-%s", s.Config.RandomIdentifier))
+	assert.Equal(s.T(), *vpc.CidrAssociations[0], "172.16.0.0/16")
+	assert.Equal(s.T(), *vpc.CidrBlock, "172.16.0.0/16")
+	assert.Nil(s.T(), vpc.Ipv6CidrAssociations)
+	assert.Equal(s.T(), vpc.Tags["Environment"], "ue2")
+	assert.Equal(s.T(), vpc.Tags["Namespace"], "eg")
+	assert.Equal(s.T(), vpc.Tags["Stage"], "test")
+	assert.Equal(s.T(), vpc.Tags["Tenant"], "default")
+
+	subnets := vpc.Subnets
+	require.Equal(s.T(), 4, len(subnets))
+
+	public_subnet_ids := atmos.OutputList(s.T(), options, "public_subnet_ids")
+	assert.Equal(s.T(), 2, len(public_subnet_ids))
+
+	public_subnet_cidrs := atmos.OutputList(s.T(), options, "public_subnet_cidrs")
+	assert.Equal(s.T(), 2, len(public_subnet_cidrs))
+
+	private_subnet_ids := atmos.OutputList(s.T(), options, "private_subnet_ids")
+	assert.Equal(s.T(), 2, len(private_subnet_ids))
+
+	private_subnet_cidrs := atmos.OutputList(s.T(), options, "private_subnet_cidrs")
+	assert.Equal(s.T(), 2, len(private_subnet_cidrs))
+
+	assert.False(s.T(), aws.IsPublicSubnet(s.T(), private_subnet_ids[0], awsRegion))
+	assert.False(s.T(), aws.IsPublicSubnet(s.T(), private_subnet_ids[1], awsRegion))
+
+	assert.True(s.T(), aws.IsPublicSubnet(s.T(), public_subnet_ids[0], awsRegion))
+	assert.True(s.T(), aws.IsPublicSubnet(s.T(), public_subnet_ids[1], awsRegion))
+
+	nats, err := GetNatsByVpcIdE(s.T(), vpcId, awsRegion)
+	assert.NoError(s.T(), err)
+	assert.Equal(s.T(), 1, len(nats))
+}
+
+func (s *ComponentSuite) TestEnabledFlag() {
+	s.T().Skip("Skipping disabled VPC test")
+	const component = "vpc/disabled"
+	const stack = "default-test"
+	s.VerifyEnabledFlag(component, stack, nil)
+}
+
+func TestRunVPCSuite(t *testing.T) {
+	suite := new(ComponentSuite)
+	helper.Run(t, suite)
+}
+
+func GetNatsByVpcIdE(t *testing.T, vpcId string, awsRegion string) ([]types.NatGateway, error) {
+	client, err := aws.NewEc2ClientE(t, awsRegion)
+	if err != nil {
+		return nil, err
+	}
+
+	filter := &types.Filter{Name: awstypes.String("vpc-id"), Values: []string{vpcId}}
+	response, err := client.DescribeNatGateways(context.Background(), &ec2.DescribeNatGatewaysInput{
+		Filter: []types.Filter{*filter},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return response.NatGateways, nil
+}
```

Comment:
consider moving this helper to the `test-helpers` repo? I think a good spot would be `pkg/aws`

Reply:
My first choice would also be contributing it back to Terratest, but they haven't merged a contribution I made over 8 months ago yet, so I don't have much hope there. I think it's likely there will be other functions that we need to complete our tests, so having our own package of helpers makes sense to me. Also, if it's not in a central place (the test-helpers lib) then no one will discover that we already have that code and will likely write that again if they they the functionality.

Reply:
Done